OMNotebook/OMNotebook/OMNotebookGUI/notebook.cpp

```diff
@@ -2526,6 +2526,8 @@ void NotebookWindow::closeEvent( QCloseEvent *event )
   QString filename = QFileInfo( subject_->getFilename() ).fileName();
   filename.remove( "\n" );
 
+  application()->clearPasteboard();
```

Comment:
I think we should do this in destructor. Assume the case when user cancels the close event.

Reply:
Well you just want to clear the pasteboard when the pointers are deleted. So the best place is in the notebook destructor instead of closeEvent.

Reply:
This means that the first place (line 517, when a cell is deleted) is also not good and should be put in the destructor of the cells.

Maybe the cells need a flag, so that they know if they are stored in pasteboard. Then the pastebin can be cleared once a cell is deleted that is marked as being in the pasteboard. Or the pasteboard should be checked if a link to the destroyed cell is contained...

I wonder why CTRL-X (cut) is working (I guess that destructor is not called) - or better: is cutting a memory leak? When will the cutted cells be destroyed?

Reply:
The best solution would probably be to copy the cells to the pasteboard rather than just storing a pointer, so that it doesn't matter what happens with the original cell. But that might be easier said than done since it seems like most classes in OMNotebook violates the [rule of three](https://en.cppreference.com/w/cpp/language/rule_of_three), and while `Cell` actually has a copy constructor it doesn't seem to be copying much.

Reply:
Maybe for now we just clear the pasteboard when the close event is accepted.

Reply:
This PR was meant to (quickly) fix the segfaults. I modified the code according to the above comments.

Can this patch be applied to the 1.19.1 branch? I'm using OMNotebook in courses and exams and would like to have this segfault fixed soon. I once encountered a complete crash of OMNotebook during an exam. The student got extra time to compensate the crash, but I would like to avoid this for future exams.

Shall I open a new issue as a reminder to overhaul the copy-cells mechanism?

Reply:
> Shall I open a new issue as a reminder to overhaul the copy-cells mechanism?

Sure.

I will merge this once Jenkins is back.